spec/task_router/workflow_builder_spec.rb

```diff
@@ -0,0 +1,387 @@
+						   }
+						],
+						\"filter_friendly_name\":\"Sales\"
+					 },
+					 {
+						\"expression\":\"type=='marketing'\",
+						\"targets\":[
+						   {
+							  \"queue\":\"WQ8c62f84b61ccfa6a333757cd508f0aae\",
+							  \"priority\": 1,
+							  \"timeout\": 120
+						   },
+						   {
+							  \"queue\":\"WQ8c62f84b61ccfa6a333757cd508f0aae\",
+							  \"priority\": 3
+						   }
+						],
+						\"filter_friendly_name\":\"Marketing\"
+					 },
+					 {
+						\"expression\":\"type=='support'\",
+						\"targets\":[
+						   {
+							  \"queue\":\"WQ5940dc0da87eaf6e3321d62041d4403b\",
+							  \"priority\": 10,
+							  \"timeout\": 30
+						   },
+						   {
+							  \"queue\":\"WQ5940dc0da87eaf6e3321d62041d4403b\",
+							  \"priority\": 15
+						   }
+						],
+						\"filter_friendly_name\":\"Support\"
+					 }
+				  ],
+				  \"default_filter\":{
+					 \"queue\":\"WQ6d29383312b24bd898a8df32779fc043\"
+				  }
+				}
+			}"
+
+      config = Twilio::TaskRouter::WorkflowConfiguration.parse_json(json)
+
+      expect(config.rules.length).to eq(3)
+      expect(config.default_target).to_not be_nil
+      # sales assertions
+      expect(config.rules[0].expression).to eq("type=='sales'")
+      expect(config.rules[0].friendly_name).to eq("Sales")
+      expect(config.rules[0].targets.length).to eq(2)
+      expect(config.rules[0].targets[0].queue).to eq("WQf6724bd5005b30eeb6ea990c3e59e536")
+      expect(config.rules[0].targets[0].priority).to eq(5)
+      expect(config.rules[0].targets[0].timeout).to eq(30)
+      expect(config.rules[0].targets[1].queue).to eq("WQf6724bd5005b30eeb6ea990c3e59e536")
+      expect(config.rules[0].targets[1].priority).to eq(10)
+      # marketing assertions
+      expect(config.rules[1].expression).to eq("type=='marketing'")
+      expect(config.rules[1].friendly_name).to eq("Marketing")
+      expect(config.rules[1].targets.length).to eq(2)
+      expect(config.rules[1].targets[0].queue).to eq("WQ8c62f84b61ccfa6a333757cd508f0aae")
+      expect(config.rules[1].targets[0].priority).to eq(1)
+      expect(config.rules[1].targets[0].timeout).to eq(120)
+      expect(config.rules[1].targets[1].queue).to eq("WQ8c62f84b61ccfa6a333757cd508f0aae")
+      expect(config.rules[1].targets[1].priority).to eq(3)
+      # support assertions
+      expect(config.rules[2].expression).to eq("type=='support'")
+      expect(config.rules[2].friendly_name).to eq("Support")
+      expect(config.rules[2].targets.length).to eq(2)
+      expect(config.rules[2].targets[0].queue).to eq("WQ5940dc0da87eaf6e3321d62041d4403b")
+      expect(config.rules[2].targets[0].priority).to eq(10)
+      expect(config.rules[2].targets[0].timeout).to eq(30)
+      expect(config.rules[2].targets[1].queue).to eq("WQ5940dc0da87eaf6e3321d62041d4403b")
+      expect(config.rules[2].targets[1].priority).to eq(15)
+      # default filter
+      expect(config.default_target.queue).to eq("WQ6d29383312b24bd898a8df32779fc043")
```

Comment:
we should also test that the same workflow is marshaled back as friendly_name instead of filter_friendly_name